### PR TITLE
Improve support for `'latex_inline'` unit format

### PR DESCRIPTION
**Pull request by eerovaher**
### Description

There are currently a handful of classes that accept the `'latex'` unit format in their `to_string()` method, but not the very similar `'latex_inline'` format:
```python
>>> u.mag(u.km/u.s).to_string('latex_inline')
    ...
ValueError: Function units cannot be written in latex_inline format. Only 'generic', 'unscaled' and 'latex' are supported.
>>> Angle(1*u.rad).to_string(format='latex_inline')
    ...
ValueError: Unknown format 'latex_inline'
>>> (1*u.km/u.s).to_string(format='latex_inline')
    ...
ValueError: Unknown format 'latex_inline'
```
This pull request enables applying the `'latex_inline'` unit format to those classes:
```python
>>> u.mag(u.km/u.s).to_string('latex_inline')
'$\\mathrm{mag}$$\\mathrm{\\left( \\mathrm{km\\,s^{-1}} \\right)}$'
>>> Angle(1*u.rad).to_string(format='latex_inline')
'$1\\mathrm{rad}$'
>>> (1*u.km/u.s).to_string(format='latex_inline')
'$1 \\; \\mathrm{km\\,s^{-1}}$'
```
`StructuredUnit` does already support `'latex_inline'`, but the output is not what might be expected based on `'latex'`:
```python
>>> u.StructuredUnit((u.km, u.km/u.s)).to_string('latex')  # One pair of '$'
'$(\\mathrm{km}, \\mathrm{\\frac{km}{s}})$'
>>> u.StructuredUnit((u.km, u.km/u.s)).to_string('latex_inline')  # Two pairs of '$'
'($\\mathrm{km}$, $\\mathrm{km\\,s^{-1}}$)'
```
This pull request updates the `StructuredUnit.to_string()` method to ensure a more consistent output for `'latex'` and `'latex_inline'`:
```python
>>> u.StructuredUnit((u.km, u.km/u.s)).to_string('latex_inline')  # One pair of '$'
'$(\\mathrm{km}, \\mathrm{km\\,s^{-1}})$'
```
### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Do the proposed changes actually accomplish desired goals?
- [ ] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [ ] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [ ] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [ ] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [ ] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [ ] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [ ] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [ ] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [ ] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

File: astropy/units/tests/test_logarithmic.py [hunk 1]
```diff
@@ -196,9 +196,17 @@ def test_str(self):
 
         lu4 = u.mag(u.ct)
         assert lu4.to_string('generic') == 'mag(ct)'
-        assert lu4.to_string('latex') == ('$\\mathrm{mag}$$\\mathrm{\\left( '
-                                          '\\mathrm{ct} \\right)}$')
-        assert lu4._repr_latex_() == lu4.to_string('latex')
+        latex_str = r'$\mathrm{mag}$$\mathrm{\left( \mathrm{ct} \right)}$'
```
Comment (mhvk) [inline astropy/units/tests/test_logarithmic.py:199]: This result actually seems rather weird, but oh well, unrelated to the PR here!

**Review by mhvk — APPROVED** [review]
Yes, nice! (-dev failure unrelated). I'll let it sit a bit so others can look if they want, but looks all good to me.

File: astropy/units/quantity.py [hunk 2]
```diff
@@ -1375,9 +1380,12 @@ def complex_formatter(value):
 
         # Format unit
         # [1:-1] strips the '$' on either side needed for math mode
-        latex_unit = (self.unit._repr_latex_()[1:-1]  # note this is unicode
-                      if self.unit is not None
-                      else _UNIT_NOT_INITIALISED)
+        if self.unit is None:
+            latex_unit = _UNIT_NOT_INITIALISED
```
Comment (eerovaher) [inline astropy/units/quantity.py:1384]: I wanted to write a test for this code path but I am unable to actually trigger it. Can `Quantity.unit` ever be `None` or should these two lines be removed?

Comment (eerovaher) [reply astropy/units/quantity.py:1384]: I've added a test for this case, so the test coverage of the patch should now be 100%. Is this good to merge or should I rebase it to fix the image tests?

**Review by nstarman — APPROVED** [review]
LGTM! No need to rebase.

**Pull request merged**